src/ProjectTemplates/Web.Spa.ProjectTemplates/content/Angular-CSharp/ClientApp/src/api-authorization/authorize.interceptor.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpEvent } from '@angular/common/http';
+import { Observable, from } from 'rxjs';
+import { AuthorizeService } from './authorize.service';
+import { mergeMap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthorizeInterceptor implements HttpInterceptor {
+  constructor(private authorize: AuthorizeService) { }
+
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return this.authorize.getAccessToken()
+      .pipe(mergeMap(token => this.processRequestWithToken(token, req, next)));
+  }
+
+  // Checks if there is an access_token available in the authorize service
+  // and adds it to the request in case it's targeted at the same origin as the
+  // single page application.
+  private processRequestWithToken(token: string, req: HttpRequest<any>, next: HttpHandler) {
+    if (!!token && this.IsSameOriginUrl(req)) {
+      req = req.clone({
+        setHeaders: {
+          Authorization: `Bearer ${token}`
+        }
+      });
+    }
+
+    return next.handle(req);
+  }
+
+  private IsSameOriginUrl(req: any) {
+    // It's an absolute url with the same origin.
+    if(req.url.startsWith(`${window.location.origin}/`)){
```

Comment:
TSLint: `missing whitespace` (both after `if` and before `{` on lines 35, 41, and 46)

Reply:
Done